### PR TITLE
docs(components): overhaul for consistency and functionality

### PR DESCRIPTION
**Pull request by CITguy**
Free Friday 2018-07-06

Preview via https://deploy-preview-281--helix-ui.netlify.com/

## Basic Correction of Positioned Element Behavior
Tooltips, Menus, and Popovers updated to use `position: fixed;` instead of `position: absolute;` and the positioning utility has been updated to calculate `(x,y)` coordinates based on the viewport, instead of the document. (Partial fix of #250)

## Massive Documentation Update

* Combine "Busy" and "Progress" into "Loaders"
* Move "Status Pills" into "Pills"
* Normalize indentation of all code snippets
* Normalize headings (no more "Demo" headings)
* Update VueJS demos to stop using deprecated functionality.
* Rearrange Syntax Highlighting demos into a tabset
* Test beta form styles in demos
* Move all visual state demos/tests out of docs and into appropriate test files.
* Add "Title" option for Choice Tiles demo
* Add interactive demos for "Text Inputs"
* Remove many unnecessary visreg tests

### LGTM's
- [x] Dev LGTM
- [x] Design LGTM
- [x] Zoom LGTM

### CODE REVIEW

File: src/helix-ui/styles/elements/hx-menu.less [hunk 1]
```diff
@@ -7,7 +7,7 @@ hx-menu {
   display: none;
   max-width: 20rem;
   min-width: 10rem;
-  position: absolute;
+  position: fixed;
```
Comment (CITguy) [inline src/helix-ui/styles/elements/hx-menu.less]: Remove commit with this fix.

File: src/helix-ui/styles/elements/hx-popover.less [hunk 2]
```diff
@@ -12,7 +12,7 @@ hx-popover {
   letter-spacing: normal;
   max-width: 32rem; //~512px
   min-width: 20rem; //~320px
-  position: absolute;
+  position: fixed;
```
Comment (CITguy) [inline src/helix-ui/styles/elements/hx-popover.less]: Remove commit with this fix.

File: src/helix-ui/styles/elements/hx-tooltip.less [hunk 3]
```diff
@@ -13,7 +13,7 @@ hx-tooltip {
     left: 0;
     letter-spacing: normal;
     max-width: 25rem;
-    position: absolute;
+    position: fixed;
```
Comment (CITguy) [inline src/helix-ui/styles/elements/hx-tooltip.less]: Remove commit with this fix.

File: src/helix-ui/util/position.js [hunk 4]
```diff
@@ -111,18 +109,11 @@ function _getCoords (position, offsetElement, referenceElement, config) {
...
-        bottom: window.innerHeight + window.pageYOffset,
-        left: window.pageXOffset,
-    };
-
     let offscreen = {
-        top: elementBox.top < viewportBox.top,
-        right: elementBox.right > viewportBox.right,
-        bottom: elementBox.bottom > viewportBox.bottom,
-        left: elementBox.left < viewportBox.left,
+        top: elementBox.top < 0,
```
Comment (CITguy) [inline src/helix-ui/util/position.js]: Remove commit with this fix.

**Comment by CITguy** [comment]
Note to self:
- [ ] make sure there's an explanation before every demo

File: docs/docs.less [hunk 5]
```diff
@@ -205,3 +199,72 @@ dl:not(.hxList) {
     justify-content: center;
   }
 }
+
+.example {
+  border: 1px solid @gray-300;
+  // a.k.a. config/options
+  > header {
+    background-color: @gray-100;
```
Comment (cathysiller) [inline docs/docs.less]: alphabetize plsss

File: docs/docs.less [hunk 6]
```diff
@@ -205,3 +199,72 @@ dl:not(.hxList) {
...
+      'config code';
+    grid-template-columns: auto 1fr;
+    grid-template-rows: 1fr auto;
+
+    > header {
+      -ms-grid-column: 1;
+      -ms-grid-row-span: 2;
+      -ms-grid-row: 1;
+      grid-area: config;
+      //margin: 0 1.25rem 0 0;
```
Comment (cathysiller) [inline docs/docs.less]: remove

**Comment by CITguy** [comment]
@cathysiller, @nimbinatus: Feedback address in https://github.com/rackerlabs/helix-ui/pull/281/commits/4462253b5622ba99d003fbb3878111a0134c1374. Ready for re-review.

File: docs/components/tabs/index.html [hunk 7]
```diff
@@ -11,99 +11,104 @@
...
-              <div class="hxCol hxSpan-6">
-                <p>
-                  Cupcake ipsum dolor sit amet bonbon topping caramels. Sesame snaps
-                  gummi bears liquorice cookie chupa chups fruitcake croissant
-                  chocolate topping. Brownie biscuit wafer marshmallow liquorice
-                  soufflé powder jelly.
-                </p>
+    <h2 id="basic-tabset">Basic Tabset</h2>
+    <div class="example">
+      <div class="tabs-demo" data-visreg="tabs">
```
Comment (nimbinatus) [inline docs/components/tabs/index.html]: Do we still need the `data-visreg` attribute here?

Comment (CITguy) [reply docs/components/tabs/index.html]: yes, because we haven't removed visreg tests yet

**Pull request merged**